sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1090,6 +1090,28 @@ class SessionCatalog(
     dbViews ++ listLocalTempViews(pattern)
   }
 
+  /**
+   * List all matching temp views in the specified database, including global/local temporary views.
+   */
+  def listTempViews(db: String, pattern: String): Seq[CatalogTable] = {
+    val dbName = format(db)
+    val globalTempViews = if (dbName == globalTempViewManager.database) {
+      globalTempViewManager.listViewNames(pattern).map { viewName =>
+        globalTempViewManager.get(viewName).map(_.tableMeta).getOrElse(
+          throw new NoSuchTableException(globalTempViewManager.database, viewName))
```

Comment:
It's weird to throw this error during listing views. Shall we use `flatMap` and just skip the temp views that were deleted immediately after `globalTempViewManager.listViewNames`?

Reply:
Okay